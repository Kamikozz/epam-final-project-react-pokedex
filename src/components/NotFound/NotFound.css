.Not-found {
  background-color: #282c34 !important;
  min-height: 100vh !important;
  display: flex !important;
  flex-direction: column !important;
  align-items: center !important;
  justify-content: center !important;
  font-size: calc(10px + 2vmin) !important;
  color: white !important;
}

.Card {
  text-align: center;
  justify-content: center;
}

.btn {
  background-color: red !important;
}