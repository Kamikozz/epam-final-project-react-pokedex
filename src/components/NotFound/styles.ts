import { createStyles } from "@material-ui/core";

const styles = createStyles({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  },
  card: {
    minWidth: 275
  },
  cardContent: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  errorStatus: {
    color: "red"
  },
  notFoundImage: {
    margin: "12px 0 12px 0",
    width: "80%",
    borderRadius: "4px"
  },
  button: {
    backgroundColor: "red",
    "&:hover": {
      backgroundColor: "red"
    }
  }
});

export default styles;
