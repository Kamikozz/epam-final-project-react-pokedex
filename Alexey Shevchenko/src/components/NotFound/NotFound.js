import React from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import './NotFound.css';
import {Link} from 'react-router-dom';

const styles = {
	card: {
		minWidth: 275,
	},
	errNum: {
		color: 'red'
	},
	errText: {
		marginBottom: 12,
	}
};

class NotFound extends React.Component {
	render() {
		const {classes} = this.props;

		return (
			<div className={'Not-found'}>
				<Card className={classes.card}>
					<CardContent className={'Card'}>
						<Typography
							variant="h2"
							className={classes.errNum}
						>
							404
						</Typography>
						<Typography
							className={classes.errText}
							color="textSecondary">
							А это что за покемон? :)
						</Typography>
						<Button
							variant="contained"
							color="primary"
							className={'btn'}
							component={Link}
							to="/">
							Go to home page
						</Button>
					</CardContent>
				</Card>
			</div>
		);
	}
}

NotFound.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(NotFound);