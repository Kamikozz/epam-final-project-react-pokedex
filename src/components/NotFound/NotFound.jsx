import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

import routes from "../../routes";
import notFoundImage from "../../assets/pokemon-meme.gif";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  },
  card: {
    minWidth: 275
  },
  cardContent: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  errorStatus: {
    color: "red"
  },
  notFoundImage: {
    margin: "12px 0 12px 0",
    width: "80%",
    borderRadius: "4px"
  },
  button: {
    backgroundColor: "red",
    "&:hover": {
      backgroundColor: "red"
    }
  }
};

class NotFound extends React.Component {
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Card className={classes.card}>
          <CardContent className={classes.cardContent}>
            <Typography variant="h2" className={classes.errorStatus}>
              404
            </Typography>
            <Typography className={classes.errorText}>
              –ê —ç—Ç–æ —á—Ç–æ –∑–∞ –ø–æ–∫–µ–º–æ–Ω? üòÖ
            </Typography>
            <img
              className={classes.notFoundImage}
              src={notFoundImage}
              alt="who is this pokemon meme"
            />
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              component={Link}
              to={routes.pokemonsList}
            >
              Go to home page
            </Button>
          </CardContent>
        </Card>
      </div>
    );
  }
}

NotFound.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(NotFound);
