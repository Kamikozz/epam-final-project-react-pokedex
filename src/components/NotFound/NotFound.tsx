import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { WithStyles, createStyles } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

import routes from "../../routes";
import notFoundImage from "../../assets/pokemon-meme.gif";

const styles = createStyles({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  },
  card: {
    minWidth: 275
  },
  cardContent: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  errorStatus: {
    color: "red"
  },
  notFoundImage: {
    margin: "12px 0 12px 0",
    width: "80%",
    borderRadius: "4px"
  },
  button: {
    backgroundColor: "red",
    "&:hover": {
      backgroundColor: "red"
    }
  }
});

interface Props extends WithStyles<typeof styles> {
  classes: {
    root: string;
    card: string;
    cardContent: string;
    errorStatus: string;
    notFoundImage: string;
    button: string;
  };
};

const LinkToPokemonsPage = (props: any) => <Link to={routes.pokemonsPage} {...props} />;
class NotFound extends React.Component<Props> {
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Card className={classes.card}>
          <CardContent className={classes.cardContent}>
            <Typography variant="h2" className={classes.errorStatus}>
              404
            </Typography>
            <Typography>
              Who's that Pokemon? ðŸ˜…
            </Typography>
            <img
              className={classes.notFoundImage}
              src={notFoundImage}
              alt="who is that pokemon meme"
            />
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              component={LinkToPokemonsPage}
            >
              Go to home page
            </Button>
          </CardContent>
        </Card>
      </div>
    );
  }
}

(NotFound as React.ComponentClass<Props>).propTypes = {
  classes: PropTypes.object.isRequired
} as any;

export default withStyles(styles)(NotFound);
