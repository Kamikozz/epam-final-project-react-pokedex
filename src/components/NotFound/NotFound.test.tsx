import React from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { render } from '@testing-library/react';

import NotFound from './NotFound';

it('renders without crashing', () => {
  render(
    <Router>
      <NotFound />
    </Router>
  );
});

it('matches snapshot', () => {
  const { container } = render(
    <Router>
      <NotFound />
    </Router>
  );
  expect(container.firstChild).toMatchSnapshot()
});

