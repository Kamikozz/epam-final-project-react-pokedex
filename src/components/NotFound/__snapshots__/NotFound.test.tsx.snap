// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="NotFound-root-108"
>
  <div
    class="MuiPaper-root-115 MuiPaper-elevation1-118 MuiPaper-rounded-116 MuiCard-root-114 NotFound-card-109"
  >
    <div
      class="MuiCardContent-root-142 NotFound-cardContent-110"
    >
      <h2
        class="MuiTypography-root-143 MuiTypography-h2-156 NotFound-errorStatus-111"
      >
        404
      </h2>
      <p
        class="MuiTypography-root-143 MuiTypography-body1-152"
      >
        Who's that Pokemon? ðŸ˜…
      </p>
      <img
        alt="who is that pokemon meme"
        class="NotFound-notFoundImage-112"
        src="pokemon-meme.gif"
      />
      <a
        class="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-contained-190 MuiButton-containedPrimary-191 MuiButton-raised-193 MuiButton-raisedPrimary-194 NotFound-button-113"
        href="/"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label-180"
        >
          Go to home page
        </span>
        <span
          class="MuiTouchRipple-root-208"
        />
      </a>
    </div>
  </div>
</div>
`;
